fix: Fix endpoints auth

andrii-balitskyi opened the pull request:
Fixes some tests in https://github.com/seamapi/javascript-http/pull/150

- **Fix /workspaces/list auth and seeded access token**
- **Add pat_with_workspace auth to /devices/get**
- **Test**
- **Add user_session and user_session_without_workspace auth, add those to /workspaces/list**
- **Add user_session test**

razor-x commented on src/pages/api/workspaces/list.ts:
```diff
@@ -5,18 +5,32 @@ import { type Workspace, workspace } from "lib/zod/index.ts"
-  )
+  let workspaces: Workspace[]
+
+  if (req.auth.type === "api_key" || req.auth.type === "client_session") {
```
Try using  `if ('workspace_id' in req.auth)`

andrii-balitskyi replied:
Hmm, even what you suggested for some reason doesn't work... This is what I come with cbf32a591f4073ba9d9f49196946c508a299e5a6

andrii-balitskyi commented:
@razor-x I also added user_session and user_session_without_workspace auth and added those to /workspaces/list auth here 5fa6413d8499c62a135f6d489d0e1b727dd1ba28. Now this PR should fix most of the failing tests in the JS SDK except for those explained in this [thread](https://hello-seam.slack.com/archives/C07DTP6EGBZ/p1731594161456569?thread_ts=1731528462.613209&cid=C07DTP6EGBZ).

razor-x commented on src/lib/middleware/with-route-spec.ts:
```diff
@@ -51,6 +61,10 @@ export const withRouteSpec = createWithRouteSpec({
     pat_with_workspace: withAccessToken({ is_workspace_id_required: true }),
     pat_without_workspace: withAccessToken({ is_workspace_id_required: false }),
     console_session: withSessionAuth({ is_workspace_id_required: true }),
+    user_session: withSessionAuth({ is_workspace_id_required: true }),
+    user_session_without_workspace: withSessionAuth({
+      is_workspace_id_required: false,
+    }),
```
What is a user session?

andrii-balitskyi replied:
I just saw them being used in [seam-connect](https://github.com/seamapi/seam-connect/blob/9b14d85d9c67c9bf8fb8b68a717b34c5b3294bc5/pages/api/public/workspaces/list.ts#L16-L17) and decided to add them. I wasn't aware those are legacy. My bad.

andrii-balitskyi replied:
Also, this [test](https://github.com/seamapi/javascript-http/pull/150/files#diff-9b4289ec4b49ae2a269c4316344c432aeb0ea3af624ca639427d89c324d65389R106-R122) uses console session auth to request /workspaces/list but the endpoint doesn't support such an auth method on [seam-connect](https://github.com/seamapi/seam-connect/blob/9b14d85d9c67c9bf8fb8b68a717b34c5b3294bc5/pages/api/public/workspaces/list.ts#L14-L19).
Should console session auth be added to /workspaces/list?

andrii-balitskyi replied:
Removed user session auth 71c02182b65dc374e03a8182d0ed5abd3c1f26f2

andrii-balitskyi replied:
@razor-x If user_session is legacy, what auth method should handle JWTs in /workspaces/list endpoint to make this [test](https://github.com/seamapi/javascript-http/pull/150/files#diff-9b4289ec4b49ae2a269c4316344c432aeb0ea3af624ca639427d89c324d65389R106-R122) pass?

razor-x commented on src/pages/api/workspaces/list.ts:
```diff
@@ -5,18 +5,40 @@ import { type Workspace, workspace } from "lib/zod/index.ts"
 
 export default withRouteSpec({
   methods: ["GET", "POST"],
-  auth: "cst_ak_pk",
```
Can we remove this from the rest of the project now?

andrii-balitskyi replied:
6cb24e1b82b9fc3c14783c2e17992a051ff9f2df
